src/main/scala/beam/sim/BeamSim.scala

```diff
@@ -291,17 +288,33 @@ class BeamSim @Inject()(
     )
 
     //rename output files generated by matsim to follow the standard naming convention of camel case
-    renameGeneratedOutputFiles(event)
+    val outputFiles = renameGeneratedOutputFiles(event)
+
+    val scenario = event.getServices.getScenario
+    val controllerIO = event.getServices.getControlerIO
+
+    writeScenario(scenario, controllerIO)
+
     outputFilesToDelete.foreach(deleteOutputFile)
 
     def deleteOutputFile(fileName: String) = {
       logger.debug(s"deleting output file: $fileName")
-      Files.deleteIfExists(Paths.get(event.getServices.getControlerIO.getOutputFilename(fileName)))
+      Files.deleteIfExists(Paths.get(controllerIO.getOutputFilename(fileName)))
     }
     BeamConfigChangesObservable.clear()
 
   }
 
+  private def writeScenario(scenario: Scenario, controlerIO: OutputDirectoryHierarchy) = {
+    PopulationCsvWriter.toCsv(scenario, controlerIO.getOutputFilename("population.csv"))
```

Comment:
we have config parameters (intervals) for deciding, when to write these out. can they be used here as well?

Reply:
as you can see in this line it is checked after every iteration ends
https://github.com/LBNL-UCB-STI/beam/pull/1804/files#diff-daac2f006f1a8b03e4c5a570b8b1ade7R189

IMO this should be responsibility of the caller. The executor should just write the scenario if asked for. It is easy to change... just tell me if I should keep the way it is or change

Reply:
yes, but there are different parameters, which control different files

Reply:
Executing before the first iteration:
```
    PopulationCsvWriter.toCsv(scenario, controllerIO.getOutputFilename("population.csv"))
    VehiclesCsvWriter(beamServices).toCsv(scenario, controllerIO.getOutputFilename("vehicles.csv"))
    HouseholdsCsvWriter.toCsv(scenario, controllerIO.getOutputFilename("households.csv"))
    NetworkCsvWriter.toCsv(scenario, controllerIO.getOutputFilename("network.csv"))
```

Executing after the end of each iteration:
```Scala
      if (interval > 0 && iterationNumber % interval == 0) {
        PlansCsvWriter.toCsv(scenario, controllerIO.getIterationFilename(iterationNumber, "plans.csv"))
      }
```